### PR TITLE
More TBD work

### PR DESCRIPTION
**Pull request by huonw**
This includes two somewhat debatable changes, both of which bias the generated TBD towards including extra symbols (i.e. ones that aren't public in the final library) instead of missing them. The logic is that extra symbols should not affect downstream users as much, whereas missing a symbol may cause linker errors.

### CODE REVIEW

**Comment by huonw** [comment]
@swift-ci Please smoke test

**Comment by huonw** [comment]
@swift-ci Please smoke test

**Comment by huonw** [comment]
@swift-ci Please smoke test and merge

File: lib/AST/Decl.cpp [hunk 1]
```diff
@@ -3356,6 +3356,28 @@ FuncDecl *AbstractStorageDecl::getAccessorFunction(AccessorKind kind) const {
...
+  auto tryPush = [&](Decl *decl) {
+    if (decl)
+      decls.push_back(decl);
+  };
+
+  tryPush(getGetter());
+  tryPush(getSetter());
+  tryPush(getMaterializeForSetFunc());
+
+  if (hasObservers()) {
```
Comment (slavapestov) [inline lib/AST/Decl.cpp]: Can observers and addressors ever be public?

File: lib/TBDGen/TBDGen.cpp [hunk 2]
```diff
@@ -281,9 +281,11 @@ void TBDGenVisitor::visitClassDecl(ClassDecl *CD) {
     auto hasFieldOffset =
         !isGeneric && var && var->hasStorage() && !var->isStatic();
     if (hasFieldOffset) {
-      // Field are only direct if the class's internals are completely known.
-      auto isIndirect = !CD->hasFixedLayout();
-      addSymbol(LinkEntity::forFieldOffset(var, isIndirect));
+      // FIXME: a field only has one sort of offset, but it is moderately
```
Comment (slavapestov) [inline lib/TBDGen/TBDGen.cpp]: Should we find a way to use IRGen type lowering from TBDGen?

Comment (huonw) [inline lib/TBDGen/TBDGen.cpp]: Yes, rdar://problem/32252689 .

File: lib/AST/Decl.cpp (hunk 1 above)
Comment (huonw) [inline lib/AST/Decl.cpp]: I don't know if it is correct for them to be public, but yes, it seems that addressors can currently: rdar://problem/32391290

**Comment by huonw** [comment]
@swift-ci Please smoke test

**Comment by huonw** [comment]
I've updated this, and the last commit especially is ugly, maybe @slavapestov has some suggestions about zipping together the functions associated with a requirement and witnessing abstract storage decl?

**Comment by huonw** [comment]
@swift-ci Please smoke test

**Comment by huonw** [comment]
@swift-ci Please smoke test

File: lib/AST/Decl.cpp (hunk 1 above)
Comment (jrose-apple) [inline lib/AST/Decl.cpp]: Observers cannot, addressors can.

**Comment by huonw** [comment]
@swift-ci Please smoke test

**Comment by huonw** [comment]
@swift-ci Please smoke test.

File: lib/AST/Decl.cpp (hunk 1 above)
Comment (slavapestov) [inline lib/AST/Decl.cpp]: Does this mean addressors are sometimes called directly, and not just via getter/setter/materializeForSet?

Comment (jrose-apple) [inline lib/AST/Decl.cpp]: Ah, maybe I'm wrong, but I thought so. @rjmccall?

**Pull request merged**